mlflow/store/sqlalchemy_store.py

```diff
@@ -161,71 +183,78 @@ def _get_experiment(self, experiment_id, view_type):
         return experiments[0]
 
     def get_experiment(self, experiment_id):
-        return self._get_experiment(experiment_id, ViewType.ALL).to_mlflow_entity()
+        with self.ManagedSessionMaker() as session:
+            return self._get_experiment(session, experiment_id, ViewType.ALL).to_mlflow_entity()
 
     def get_experiment_by_name(self, experiment_name):
         """
         Specialized implementation for SQL backed store.
         """
-        experiments = self._list_experiments(names=[experiment_name], view_type=ViewType.ALL).all()
-        if len(experiments) == 0:
-            return None
+        with self.ManagedSessionMaker() as session:
+            experiments = self._list_experiments(
+                names=[experiment_name], view_type=ViewType.ALL, session=session).all()
+            if len(experiments) == 0:
+                return None
 
-        if len(experiments) > 1:
-            raise MlflowException('Expected only 1 experiment with name={}. Found {}.'.format(
-                experiment_name, len(experiments)), INVALID_STATE)
+            if len(experiments) > 1:
+                raise MlflowException('Expected only 1 experiment with name={}. Found {}.'.format(
+                    experiment_name, len(experiments)), INVALID_STATE)
 
-        return experiments[0]
```

Comment:
@mparkhe Looks like the public `get_experiment_by_name()` function was previously returning a raw `SqlExperiment` type. The `test_get_experiment` method caught this because the returned `SqlExperiment` lost its session scope after `get_experiment_by_name()` returned. This provides some validation for our reasoning that scoped sessions help avoid erroneous behavior :D!

Reply:
Whoa!!